vnext/Shared/Utils.cpp

```diff
@@ -4,10 +4,49 @@
 #include "Utils.h"
 #include <regex>
 
+#include <DesktopWindowBridge.h>
+#include <ShlObj.h>
+#include <Shlwapi.h>
+#include <appmodel.h>
+#include <winrt/Windows.Storage.h>
+
 using std::string;
 
 namespace Microsoft::React {
 
+std::future<std::string> getApplicationDataPath(const wchar_t *childFolder) {
+  uint32_t length{0};
+  bool isPackagedApp = GetCurrentPackageFullName(&length, nullptr) != APPMODEL_ERROR_NO_PACKAGE;
+  std::string appDataPath;
+  if (isPackagedApp) {
+    if (!childFolder) {
+      co_return winrt::to_string(winrt::Windows::Storage::ApplicationData::Current().LocalFolder().Path());
```

Comment:
Should we lift the expression up to `LocalFolder()` to a variable since it is shared between both branches?